### PR TITLE
284 yamlsupport fix empty uri and implement doucle quotes inside uri

### PR DESCRIPTION
**Pull request by apupier**

### CODE REVIEW

File: src/main/java/com/github/cameltooling/lsp/internal/parser/CamelKYamlDSLParser.java [hunk 1]
```diff
@@ -39,14 +39,19 @@
 	@Override
 	public String getCamelComponentUri(String line, int characterPosition) {
 		String camelComponentURI = null;
-		Map data = parseYaml(line);
-		if (data.containsKey(URI_KEY) && URI_KEY.length() < characterPosition) {
-			camelComponentURI = data.get(URI_KEY).toString();
-		} else if(data.containsKey(TO_KEY) && data.get(TO_KEY).toString().trim().length()>0) {
-			camelComponentURI = data.get(TO_KEY).toString();
+		String trLine = line.trim();
```
Comment (lhein) [inline src/main/java/com/github/cameltooling/lsp/internal/parser/CamelKYamlDSLParser.java]: I switched to SnakeYAML to get rid of mangeling the string myself.

File: src/main/java/com/github/cameltooling/lsp/internal/parser/CamelKYamlDSLParser.java [hunk 2]
```diff
@@ -60,8 +65,7 @@ public CamelURIInstance createCamelURIInstance(TextDocumentItem textDocumentItem
 
 	private int getStartCharacterInDocumentOnLinePosition(TextDocumentItem textDocumentItem, Position position) {
 		String line = parserFileHelperUtil.getLine(textDocumentItem, position.getLine());
-		String uri = extractUriFromYamlData(line);
-		return line.indexOf(uri);
+		return line.indexOf('"') + 1;
```
Comment (lhein) [inline src/main/java/com/github/cameltooling/lsp/internal/parser/CamelKYamlDSLParser.java]: this doesn't work if your uri is not enclosed in "

Comment (apupier) [reply src/main/java/com/github/cameltooling/lsp/internal/parser/CamelKYamlDSLParser.java]: yes, it wasn't working either with previous state.

File: src/test/java/com/github/cameltooling/lsp/internal/CamelLanguageServerTest.java [hunk 3]
```diff
@@ -142,47 +145,78 @@ public void testProvideCompletionForGroovyOnRealFileWithCamelKCloseToModeline()
...
-	@Test
-	public void testProvideCompletionForYamlOnRealFileWithCamelKCloseToModeline() throws Exception {
-		File f = new File("src/test/resources/workspace/samplewithModelineLike.yaml");
-		assertThat(f).exists();
-		try (FileInputStream fis = new FileInputStream(f)) {
-			CamelLanguageServer cls = initializeLanguageServer(fis, ".yaml");
-			CompletableFuture<Either<List<CompletionItem>, CompletionList>> completions = getCompletionFor(cls, new Position(12, 16));
-			assertThat(completions.get().getLeft()).contains(createExpectedAhcCompletionItem(12, 16, 12, 22));
+	@Nested
+	class Yaml {
```
Comment (lhein) [inline src/test/java/com/github/cameltooling/lsp/internal/CamelLanguageServerTest.java]: is that just for grouping all yaml based tests? maybe its better to extract it into its on test class file then?

File: src/main/java/com/github/cameltooling/lsp/internal/parser/CamelKYamlDSLParser.java (hunk 1 above)
Comment (apupier) [reply src/main/java/com/github/cameltooling/lsp/internal/parser/CamelKYamlDSLParser.java]: the problem is that SnakeYaml is returning the unescaped string with which we cannot work correctly with positions

File: src/test/java/com/github/cameltooling/lsp/internal/CamelLanguageServerTest.java (hunk 3 above)
Comment (apupier) [reply src/test/java/com/github/cameltooling/lsp/internal/CamelLanguageServerTest.java]: > is that just for grouping all yaml based tests?

yes, it is just for grouping all yaml based tests.

> maybe its better to extract it into its on test class file then?

I preferred provided the PR suggestion like that just to ease review but keeping facility to relaunch a subset of the tests while developing. I think that it is easier to see the added test cases compared to having created the class. Agree that extracted in its own class is better on long run.

Comment (lhein) [reply src/test/java/com/github/cameltooling/lsp/internal/CamelLanguageServerTest.java]: Yeah, while working on a generic fix for the escape char issue I found that notation quite useful. Will keep it as is.

Comment (lhein) [reply src/test/java/com/github/cameltooling/lsp/internal/CamelLanguageServerTest.java]: Actually I feel that if we start to extract language related tests to own files we will end up to extract the other languages as groovy / js / etc too which might in the current state be a bit over the edge.

**Comment by lhein** [comment]
I think I found a more generic solution. Closing this one.

**Pull request closed without merging**